src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java

```diff
@@ -826,6 +834,23 @@ private XSDComplexTypeDefinition buildComplexSchemaContent(
                 XSDElementDeclaration element = factory.createXSDElementDeclaration();
                 element.setName(attribute.getLocalName());
                 element.setNillable(attribute.isNillable());
+                particle = factory.createXSDParticle();
+                particle.setMinOccurs(attribute.getMinOccurs());
+                particle.setMaxOccurs(attribute.getMaxOccurs());
+                particle.setContent(element);
+                group.getContents().add(particle);
+                // if the attributeDescriptor description is populated, use it as the annotation
+                if (attribute.getType() != null && attribute.getType().getDescription() != null) {
```

Comment:
attribute.getType() is called many times in this method, I'd give it its own variable.
Also, we should never have an attribute descriptor without type, and if we did, the code would NPE right after this if block, at ``attribute.getType().getName()`` (line 855 in the PR version of the code).

Reply:
Fixed